src/manifold/src/quickhull.h

```diff
@@ -126,6 +126,7 @@ class MeshBuilder {
     size_t visibilityCheckedOnIteration = 0;
     std::uint8_t isVisibleFaceOnCurrentIteration : 1;
     std::uint8_t inFaceStack : 1;
+    std::recursive_mutex* faceMutex = new std::recursive_mutex();
```

Comment:
And this probably should not be a recursive mutex.

Reply:
I saw recursive_mutex being used at a lot of places so just figured I could use it, I'll use std::mutex instead?

Reply:
Recursive mutex is needed only when you may lock the same thing several times in the same thread. Yeah, you can probably just use `std::mutex`.